backend/controllers/userController.js

```diff
@@ -1,21 +1,22 @@
 const jwt = require("jsonwebtoken");
 const bcrypt = require("bcryptjs");
 const User = require("../models/userModel");
-const Createtodo = require("../models/taskModel");
 
-const token = (id) => {
-  return jwt.sign({ id }, process.env.JWT_KEY, { expiresIn: "20d" });
+const token = (user) => {
+  return jwt.sign({ id: user._id, role: user.role }, process.env.JWT_KEY, {
+    expiresIn: "20d",
```

Comment:
please set this env and good practice is to set expireIn only 1 day

Reply:
resolved.